### PR TITLE
PWN-5120 - show restore when needed on logout except of start screen

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5120

## Description of Work

Show restore when needed on logout except of start screen

### CODE REVIEW

**Review by davran1001 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/AuthModule.kt [hunk 1]
```diff
@@ -91,7 +91,7 @@ object AuthModule {
 
         single { BiometricManager.from(androidContext()) }
 
-        factory { AuthInteractor(get(), get(), get(), get(), get()) }
+        factory { AuthInteractor(get(), get(), get(), get(), get(), get()) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/AuthModule.kt:94]: factoryOf

File: app/src/main/java/org/p2p/wallet/auth/interactor/AuthInteractor.kt [hunk 2]
```diff
@@ -145,4 +148,6 @@ class AuthInteractor(
     }
 
     fun isFingerprintEnabled(): Boolean = getBiometricStatus() == BiometricStatus.ENABLED
+
+    fun finishSignUp() = accountStorage.remove(KEY_IN_SIGN_UP_PROCESS)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/AuthInteractor.kt:152]:
```suggestion
    fun finishSignUp() { 
        accountStorage.remove(KEY_IN_SIGN_UP_PROCESS)
    }
```

File: app/src/main/java/org/p2p/wallet/auth/repository/UserSignUpDetailsStorage.kt [hunk 3]
```diff
@@ -51,4 +54,6 @@ class UserSignUpDetailsStorage(
     }
 
     fun isDeviceShareSaved(): Boolean = getLastSignUpUserDetails() != null
+
+    fun isSignUpProcess(): Boolean = accountStorage.getString(KEY_IN_SIGN_UP_PROCESS) != null
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/UserSignUpDetailsStorage.kt:58]: isSignUpInProcess
inSignUpProcess is kinda incorrect English phrase

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt [hunk 4]
```diff
@@ -32,13 +36,19 @@ class CommonRestoreFragment :
...
+        )
+
+        fun createWithoutBack(): CommonRestoreFragment = create(showBackButton = false)
     }
 
     override val presenter: CommonRestoreContract.Presenter by inject()
 
     private val binding: FragmentCommonRestoreBinding by viewBinding()
 
+    private val useBackButton: Boolean by args(ARG_DO_NOT_SHOW_BACK)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt:50]: showBackButton, not `use`

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**